lib/once.rb

```diff
@@ -41,5 +41,16 @@ def do(name:, params:, within: DEFAULT_TIME, &block)
         block.call
       end
     end
+
+    def key_in_use?(name:, params:)
+      time_to_expire_seconds(name: name, params: params) > 0
+    end
+
+    def time_to_expire_seconds(name:, params:)
```

Comment:
ttl_seconds would be fine also, that's a pretty common term. also I would take out the hexdigest bit so that's only happening in one spot